File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
    const ans = new Map();
    ans.set(nums[0], 0)
    for (let i = 1; i < nums.length; i++) {
        if (ans.has(nums[i]))
            return true;
        ans.set(nums[i], i);
    }
    return false;
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        if (nums.length <= 1) {
            return false;
        }
        Set<Integer> values = new HashSet<Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (values.contains(nums[i])) {
                return true;
            }
            values.add(nums[i]);
        }
        return false;
    }
}